src/xfer/mod.rs

```diff
@@ -0,0 +1,144 @@
+use std::collections::HashMap;
+
+use serde::Deserialize;
+use snafu::{ensure, ResultExt, Snafu};
+use url::Url;
+
+mod cache;
+pub use cache::*;
```

Comment:
Does this need to be `pub`? If so, why not make the module itself public instead of reexporting it?